lib/galaxy/util/__init__.py

```diff
@@ -1058,7 +1058,7 @@ def smart_str(s, encoding=DEFAULT_ENCODING, strings_only=False, errors='strict')
 
 def strip_control_characters(s):
     """Strip unicode control characters from a string."""
-    return "".join(c for c in unicodify(s) if unicodedata.category(c)[0] != "C")
+    return "".join(c for c in unicodify(s) if unicodedata.category(c)[0] != "C" or unicodedata.category(c) == "Cn")
```

Comment:
Had to look twice at this to understand that `unicodedata.category(c)[0]` and `unicodedata.category(c)` are not the same.
On a sidenote it also looks like this is only an issue on python2: `unicodedata.category('💩')` is `So` on python 3.
How about this?
```suggestion
    return "".join(c for c in unicodify(s) if unicodedata.category(c) != "Cc")
```
That handles the one case I wanted to deal with in https://github.com/galaxyproject/galaxy/commit/366701cfb23d8de31eac3c4312c6accca2e74ef1

Reply:
No concerns for Cf? I couldn't quite tell what it was (their documentation for "what is in the group" is rubbish. I'm happy to just make an exception for Cn since I know that's a high range / private use area.

Reply:
I think it's only NUL (0x00) characters that are problematic, as far as I understand it might even be restricted to `\0`

Reply:
Ok, sounds good, I'll just apply your patch then.